### PR TITLE
[Python API] Add more api for parameter and result ops

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - Add setters/getters for partial_shape, element_type and layout for Parameter and Result operations

### Tickets:
 - 71499

### CODE REVIEW

File: runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp [hunk 1]
```diff
@@ -35,4 +35,22 @@ void regclass_graph_op_Parameter(py::module m) {
     parameter.def("get_partial_shape",
                   (ov::PartialShape & (ov::op::v0::Parameter::*)()) & ov::op::v0::Parameter::get_partial_shape);
     parameter.def("set_partial_shape", &ov::op::v0::Parameter::set_partial_shape);
+
+    parameter.def("get_element_type", &ov::op::v0::Parameter::get_element_type);
+
+    parameter.def("set_element_type", &ov::op::v0::Parameter::set_element_type);
```
Comment (ilya-lavrenov) [inline runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp:41]: @ilyachur why do we need set for parameter?

Comment (akuporos) [reply runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp:41]: сс @eaidova

Comment (ilyachur) [reply runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp:41]: In general this method allows to change the original model precision.

File: runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp [hunk 2]
```diff
@@ -35,4 +35,22 @@ void regclass_graph_op_Parameter(py::module m) {
...
+
+    parameter.def("get_layout", &ov::op::v0::Parameter::get_layout);
+
+    parameter.def("set_layout", &ov::op::v0::Parameter::set_layout);
+
+    parameter.def_property("partial_shape",
```
Comment (ilyachur) [inline runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp:47]: @jiwaszki Do we need to have both approaches with getters/setters and properties?

File: runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp (hunk 1 above)
Comment (eaidova) [reply runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp:41]: we need this for some validation cases (reduce memory required for model compilation for VPU), also input precision settings used in some demos for unification handling input/output data

File: runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp (hunk 2 above)
Comment (akuporos) [reply runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp:47]: The pythonic way is use of properties but we kept getters/setters to be aligned with cpp

File: runtime/bindings/python/src/pyopenvino/graph/ops/result.cpp [hunk 3]
```diff
@@ -18,4 +18,14 @@ void regclass_graph_op_Result(py::module m) {
...
 
     result.doc() = "openvino.impl.op.Result wraps ov::op::v0::Result";
+
+    result.def("get_output_partial_shape", &ov::Node::get_output_partial_shape, py::arg("index"));
+
+    result.def("get_output_element_type", &ov::Node::get_output_element_type, py::arg("index"));
+
+    result.def("get_layout", &ov::op::v0::Result::get_layout);
+
+    result.def("set_layout", &ov::op::v0::Result::set_layout);
```
Comment (jiwaszki) [inline runtime/bindings/python/src/pyopenvino/graph/ops/result.cpp:28]: Add `args` as py::arg.

File: runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp [hunk 4]
```diff
@@ -35,4 +35,22 @@ void regclass_graph_op_Parameter(py::module m) {
...
                   (ov::PartialShape & (ov::op::v0::Parameter::*)()) & ov::op::v0::Parameter::get_partial_shape);
     parameter.def("set_partial_shape", &ov::op::v0::Parameter::set_partial_shape);
+
+    parameter.def("get_element_type", &ov::op::v0::Parameter::get_element_type);
+
+    parameter.def("set_element_type", &ov::op::v0::Parameter::set_element_type);
+
+    parameter.def("get_layout", &ov::op::v0::Parameter::get_layout);
+
+    parameter.def("set_layout", &ov::op::v0::Parameter::set_layout);
```
Comment (jiwaszki) [inline runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp:45]: Missing args.

File: runtime/bindings/python/tests/test_ngraph/test_basic.py [hunk 5]
```diff
@@ -24,6 +24,11 @@ def test_ngraph_function_api():
     parameter_b = ops.parameter(shape, dtype=np.float32, name="B")
     parameter_c = ops.parameter(shape, dtype=np.float32, name="C")
     model = (parameter_a + parameter_b) * parameter_c
+
+    assert parameter_a.element_type == Type.f32
+    assert parameter_a.partial_shape == PartialShape([2, 2])
+    parameter_a.layout = ov.Layout("NCWH")
+    assert parameter_a.layout.to_string() == "[N,C,W,H]"
```
Comment (jiwaszki) [inline runtime/bindings/python/tests/test_ngraph/test_basic.py:31]: Note: compare to value?

File: runtime/bindings/python/tests/test_ngraph/test_basic.py [hunk 6]
```diff
@@ -44,7 +49,12 @@ def test_ngraph_function_api():
     assert list(function.get_output_shape(0)) == [2, 2]
     assert (function.get_parameters()[1].get_partial_shape()) == PartialShape([3, 4, 5])
     assert len(function.get_parameters()) == 3
-    assert len(function.get_results()) == 1
+    results = function.get_results()
+    assert len(results) == 1
+    assert results[0].get_output_element_type(0) == Type.f32
+    assert results[0].get_output_partial_shape(0) == PartialShape([2, 2])
+    results[0].layout = ov.Layout("NC")
+    assert results[0].layout.to_string() == "[N,C]"
```
Comment (jiwaszki) [inline runtime/bindings/python/tests/test_ngraph/test_basic.py:57]: Note: compare to value?

**Review by jiwaszki — CHANGES_REQUESTED** [review]

**Review by eaidova — APPROVED** [review]
From user point of view, everything is ok

File: runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp (hunk 4 above)
Comment (akuporos) [reply runtime/bindings/python/src/pyopenvino/graph/ops/parameter.cpp]: fixed

File: runtime/bindings/python/src/pyopenvino/graph/ops/result.cpp (hunk 3 above)
Comment (akuporos) [reply runtime/bindings/python/src/pyopenvino/graph/ops/result.cpp]: fixed

File: runtime/bindings/python/tests/test_ngraph/test_basic.py (hunk 6 above)
Comment (akuporos) [reply runtime/bindings/python/tests/test_ngraph/test_basic.py]: fixed

File: runtime/bindings/python/tests/test_ngraph/test_basic.py (hunk 5 above)
Comment (akuporos) [reply runtime/bindings/python/tests/test_ngraph/test_basic.py]: fixed

**Review by jiwaszki — APPROVED** [review]

**Pull request merged**